### ISSUE TITLE
Discrepancies with new Open Subtitles API Results When Using IINA

### ISSUE THREAD

**Issue by tsegreto** [iina/iina#4749]
I've encountered a significant issue when utilizing the new API through IINA. The results returned by the new API deviate substantially from those provided by the old API. As an illustrative example, when I input 'The.Sopranos.S06E14.Stage.5.720p.WEB-DL.h.264.DD5.1-NTb.mkv', the API returns the likes of 'Sopranos 1x01', which is clearly inaccurate. This discrepancy raises questions about the source of the issue. It's unclear whether the problem lies within the IINA integration, the new API itself, or potentially changes in the Open Subtitles libraries. Further investigation or insights into this matter would be greatly appreciated.

**System and IINA version:**

- macOS 14.2.1 (23C71)
- IINA 1.3.4 Build 140

**Comment by low-batt** [comment]
Reproduced for me. Thanks for reporting this.

I looked into the Open Subtitles REST API documentation again. The [Search for subtitles](https://opensubtitles.stoplight.io/docs/opensubtitles-api/a172317bd5ccc-search-for-subtitles) section does not mention the API [Guessit](https://opensubtitles.stoplight.io/docs/opensubtitles-api/7783e082edcf7-guessit). I believe the intent is for the client to use this API to obtain additional information to pass when calling the search API. I will give this API a try and see if it can be used to fix this issue.

**Comment by browser-bug** [comment]
I'm reporting here as well as stated in [#4678](https://github.com/iina/iina/pull/4678#issuecomment-1879625346). This is not very precise as the older version. As @low-batt said it is related to the new data expected by the Rest API. In my particular case I'm trying to download subs from TV episodes (Parks and Recreation for more context - _"Parks and Recreation (2009) - S05E02 - Soda Tax (1080p AMZN WEBRip x265 Silence).mkv"_ ).

**Comment by low-batt** [comment]
@browser-bug Very helpful that you provided this second example. I thought it was the same problem shown by the example @tsegreto supplied. Turns out there is a second issue lurking.

As seen here using the [Guessit](https://opensubtitles.stoplight.io/docs/opensubtitles-api/7783e082edcf7-guessit) API works to narrow the search results:

![WithGuessit](https://github.com/iina/iina/assets/86170219/aad99e35-352e-4f93-a380-26b207d6994e)

I thought that would also correct the Parks and Recreation example, but it does not. If you search [opensubtitles.org](https://www.opensubtitles.org/) in the browser for this:
"Parks and Recreation (2009) - S05E02 - Soda Tax (1080p AMZN WEBRip x265 Silence).mkv"

Open Subtitles returns two results. But if you search [opensubtitles.com](https://www.opensubtitles.com/) Open Subtitles does not return any matches.

It appears the .org site supports using the first year of the series and the .com site does not. If you rename the filename to use the year of the episode:
"Parks and Recreation (2012) - S05E02 - Soda Tax (1080p AMZN WEBRip x265 Silence).mkv"

Then the .com site returns 5 matches.

I have reported this to Open Subtitles. My post is currently awaiting approval before it will be shown to the public.

**Comment by low-batt** [comment]
The fix for this issue has been merged into the IINA `develop` branch. GitHub [automatically closed](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) the linked issue in reaction to the merge. I am reopening this issue until the fix is available in an official release of IINA so that users intending to report this problem can easily locate this existing report. Once the fix for this issue has been included in an official release this issue will be closed.

The IINA 1.3.4 release was rushed out due to the end of year deadline to convert to the Open Subtitles REST API. Many tasks planned for 1.3.4 were left undone. We are continuing to work on the issues that missed the deadline and are planning a follow up release with those fixes along with the changes for this issue.

On the "no results" problem…

The topic I posted in the Open Subtitles forum regarding search not returning results for "Parks and Recreation (2009) - S05E02 - Soda Tax (1080p AMZN WEBRip x265 Silence).mkv" is: [Search of .com returns no results, .org returns results](https://forum.opensubtitles.org/viewtopic.php?f=1&t=18453&sid=65f54e14c79bd6b7277d288a6330b3df)

No response as of yet.

**Comment by low-batt** [comment]
In [this comment](https://github.com/iina/iina/issues/3920#issuecomment-1888373957) Open Subtitles indicated:
> there was a temporary bug on our guessit system, it didn't detect the season & episodes properly for a few weeks.
>
> it's all fixed now, you can send your queries like "the sopranos s02e04" it'll be exactly the same as passing the season and episoode numbers separately

I've confirmed that IINA 1.3.4 is now returning subtitle search results that are for the specific episode. I have reverted the change I made on the client side to fix the problem.

On the other problem, where a search for "Parks and Recreation (2009) - S05E02 - Soda Tax (1080p AMZN WEBRip x265 Silence).mkv" returns results if [opensubtitles.org](https://www.opensubtitles.org/) is used, but searching [opensubtitles.com](https://www.opensubtitles.com/) does not return any matches. Open Subtitles has indicated they are still porting problematic subtitles from the .org site to the .com site.

As the Open Subtitles server issue has been fixed, I am going to close this issue.
